crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -1416,3 +1464,33 @@ export.exec_kernel_proc
     dynexec
     # => [<procedure_outputs>, <pad>]
 end
+
+#! Returns the memory pointer to the input note, depending on whether the requested note is current
+#! or it was requested by index.
+#!
+#! Inputs:  [is_current_note, note_index]
+#! Outputs: [input_note_ptr]
+#!
+#! Where:
+#! - is_current_note is the boolean flag indicating whether we should return requested data from
+#!   the currently processing note or from the note with the specified index.
+#! - note_index is the index of the input note whose data should be returned. Notice that if
+#!   is_current_note is 1, note_index is ignored.
+#! - input_note_ptr is the pointer to the correct input note. 
```

Comment:
If `is_current_note == 1` but we are not executing a note, `note_ptr` will be `0`, right? If so, I'd mention this in the comment.